cmd/agent/api/agent/agent.go

```diff
@@ -278,6 +281,40 @@ func getRuntimeConfig(w http.ResponseWriter, r *http.Request) {
 	w.Write(runtimeConfig)
 }
 
+func setRuntimeConfig(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	setting := vars["setting"]
+	log.Info("Got a request to change a setting: %s", setting)
+	r.ParseForm()
+	value := html.UnescapeString(r.Form.Get("value"))
+
+	if _, ok := config.RuntimeSettings[setting]; !ok {
+		body, _ := json.Marshal(map[string]string{"error": "unknown setting"})
+		http.Error(w, string(body), 500)
+		return
+	}
+	if err := config.RuntimeSettings[setting].Set(value); err != nil {
+		body, _ := json.Marshal(map[string]string{"error": err.Error()})
+		http.Error(w, string(body), 500)
```

Comment:
Ditto.

Reply:
Actually, nope. 500 is ok here.

Reply:
If an error occurs here, the setting exist but changing it failed, thus I think returning a 500 code is correct.